metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -563,9 +564,24 @@ def has_failed(self):
     @property
     def is_running(self):
         # Returns true if the pod is running.
-        return not self.is_done and (
-            self._pod.get("status", {}).get("phase") == "Running"
-        )
+        if self.is_done:
+            return False
+
+        container_statuses = self._pod.get("status", {}).get("container_statuses", [])
+        if not container_statuses:
+            # Yet to start, most likely
+            return True
+        if len(container_statuses) < 1:
+            # Yet to start, most likely
+            return True
+        # There is guaranteed to be only 1 container
+        cstatus = container_statuses[0]
```

Comment:
can we fetch the container status using the name of the container? that will guard against the scenario if/when we have more than one container.

Reply:
name of the container is derived from the step name and is not available in the `RunningJob`. 

I'll change the code to iterate over all containers and if any one of them is NOT terminated, return True for `is_running`. That should protect against the case where there are multiple containers.